libs/net/nodenetworkconfigurationpolicy.py

```diff
@@ -0,0 +1,159 @@
+# API: https://nmstate.io/devel/yaml_api.html#dhcp
+from dataclasses import asdict, dataclass
+from typing import Any
+
+from ocp_resources.exceptions import NNCPConfigurationFailed
+from ocp_resources.node_network_configuration_policy_latest import NodeNetworkConfigurationPolicy
+from ocp_resources.resource import Resource, ResourceEditor
+from timeout_sampler import retry
+
+WAIT_FOR_STATUS_TIMEOUT = 90
+WAIT_FOR_STATUS_SLEEP = 5
+
+
+@dataclass
+class STP:
+    enabled: bool
+
+
+@dataclass
+class BridgeOptions:
+    stp: STP
+
+
+@dataclass
+class Port:
+    name: str
+
+
+@dataclass
+class Bridge:
+    options: BridgeOptions | None = None
+    port: list[Port] | None = None
+    allow_extra_patch_ports: bool | None = None
+
+
+@dataclass
+class IP:
+    dhcp: bool
+    auto_dns: bool = True
+    enabled: bool = False
+
+
+@dataclass
+class IPv4(IP):
+    pass
+
+
+@dataclass
+class IPv6(IP):
+    autoconf: bool = False
+
+
+@dataclass
+class Interface:
+    name: str
+    type: str
+    state: str
+    ipv4: IPv4
+    ipv6: IPv6
+    bridge: Bridge | None = None
+
+
+@dataclass
+class DesiredState:
+    interfaces: list[Interface]
+
+
+class NNCPSuccessStatusNotMet(Exception):
+    pass
+
+
+class NetLibsNodeNetworkConfigurationPolicy(NodeNetworkConfigurationPolicy):
```

Comment:
The `NetLibs` prefix is not good and make no sense to me.

I see two options:
- Alias the import:
`from ocp_resources.node_network_configuration_policy_latest import NodeNetworkConfigurationPolicy as nncp`
- Call it `nncp`

Reply:
flake8 is wrong, but no point in arguing with a linter in this case.